docs/LoongArch-ELF-ABI-CN.adoc

```diff
@@ -647,6 +647,203 @@ v1.00
+
+|76
+|R_LARCH_GOT_PC_LO12
+|32位GOT相对PC偏移的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+G) [11 ... 0]+`
+
+|77
+|R_LARCH_GOT64_PC_LO20
+|64位GOT相对PC偏移的[32:51]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+G - (PC & ~0xffffffff)) [51 ... 32]+`
+
+|78
+|R_LARCH_GOT64_PC_HI12
+|64位GOT相对PC偏移的[52:63]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+G - (PC & ~0xffffffff)) [63 ... 52]+`
+
+|79
+|R_LARCH_GOT64_HI20
+|32位GOT绝对地址的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+G) [31 ... 12]+`
+
+|80
+|R_LARCH_GOT64_LO12
+|32位GOT绝对地址的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+G) [11 ... 0]+`
+
+|81
+|R_LARCH_GOT64_LO20
+|64位GOT绝对地址的[32:51]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+G) [51 ... 32]+`
+
+|82
+|R_LARCH_GOT64_HI12
+|64位GOT绝对地址的[52:63]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+G) [63 ... 52]+`
+
+|83
+|R_LARCH_TLS_LE_HI20
+|TLS LE 符号相对 TP 寄存器偏移的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = T [31 ... 12]+`
+
+|84
+|R_LARCH_TLS_LE_LO12
+|TLS LE 符号相对 TP 寄存器偏移的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = T [11 ... 0]+`
+
+|85
+|R_LARCH_TLS_LE64_LO20
+|TLS LE 符号相对 TP 寄存器偏移的[32:51]位
+|`(*(uint32_t *) PC) [24 ... 5] = T [51 ... 32]+`
+
+|86
+|R_LARCH_TLS_LE64_HI12
+|TLS LE 符号相对 TP 寄存器偏移的[52:63]位
+|`(*(uint32_t *) PC) [21 ... 10] = T [63 ... 52]+`
+
+|87
+|R_LARCH_TLS_IE_PC_HI20
+|TLS IE 符号 GOT 相对 PC 32位偏移的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (((GP+IE) & ~0xfff) - (PC & ~0xfff)) [31 ... 12]+`
+
+|88
+|R_LARCH_TLS_IE_PC_LO12
+|TLS IE 符号 GOT 相对 PC 32位偏移的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+IE) [11 ... 0]+`
+
+|89
+|R_LARCH_TLS_IE64_PC_LO20
+|TLS IE 符号 GOT 相对 PC 64位偏移的[32:51]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+IE - (PC & ~0xffffffff)) [51 ... 32]+`
+
+|90
+|R_LARCH_TLS_IE64_PC_HI12
+|TLS IE 符号 GOT 相对 PC 64位偏移的[52:63]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+IE - (PC & ~0xffffffff)) [63 ... 52]+`
+
+|91
+|R_LARCH_TLS_IE64_HI20
+|TLS IE 符号 GOT 绝对地址的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+IE) [31 ... 12]+`
+
+|92
+|R_LARCH_TLS_IE64_LO12
+|TLS IE 符号 GOT 绝对地址的[0:11]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+IE) [11 ... 0]+`
+
+|93
+|R_LARCH_TLS_IE64_LO20
+|TLS IE 符号 GOT 绝对地址的[32:51]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+IE) [51 ... 32]+`
+
+|94
+|R_LARCH_TLS_IE64_HI12
+|TLS IE 符号 GOT 绝对地址的[52:63]位
+|`(*(uint32_t *) PC) [21 ... 10] = (GP+IE) [63 ... 52]+`
+
+|95
+|R_LARCH_TLS_LD_PC_HI20
+|TLS LD 符号 GOT 相对 PC 32位偏移的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (((GP+GD) & ~0xfff) - (PC & ~0xfff)) [31 ... 12]+`
+
+|96
+|R_LARCH_TLS_LD64_HI20
+|TLS LD 符号 GOT 绝对地址的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+IE) [31 ... 12]+`
+
+|97
+|R_LARCH_TLS_GD_PC_HI20
+|TLS GD 符号 GOT 相对 PC 32位偏移的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (((GP+GD) & ~0xfff) - (PC & ~0xfff)) [31 ... 12]+`
+
+|98
+|R_LARCH_TLS_LD64_HI20
+|TLS GD 符号 GOT 绝对地址的[12:31]位
+|`(*(uint32_t *) PC) [24 ... 5] = (GP+IE) [31 ... 12]+`
+
+|99
+|R_LARCH_32_PCREL
+|32位相对 PC 偏移
+|`(*(uint32_t *) PC) = (S+A-PC) [31 ... 0]+`
+
+|100
+|R_LARCH_RELAX
+|暂未实现
```

Comment:
Even if it's not implemented, you should give it a definition.  Otherwise you should just leave it undefined.

Reply:
This relocation is expected to do some optimizations, do or not won't affect correctness.
The linker has implemented some specific instructions, but the assembler still has some support that has not been completed. 
If it is deleted, the current version linker can not to link next version object (maybe soon to be patched).

Reply:
Then maybe a simple description like "mark a instruction suitable for link-time relaxation" or something (does @xen0n have a better idea?)

Reply:
BTW: make sure you all read MaskRay's [excellent](http://maskray.me/blog/2021-03-14-the-dark-side-of-riscv-linker-relaxation) [writeups](http://maskray.me/blog/2022-07-10-riscv-linker-relaxation-in-lld) on the topic of linker relaxations before implementing any of these. ;-)